Source/Interceptor.cs

```diff
@@ -75,56 +74,50 @@ internal void VerifyAll()
 
 		private void VerifyOrThrow(Func<IProxyCall, bool> match)
 		{
-			lock (calls)
+			var failures = new List<IProxyCall>();
+
+			// The following verification logic will remember each processed setup so that duplicate setups
+			// (that is, setups overridden by later setups with an equivalent expression) can be detected.
+			// To speed up duplicate detection, they are partitioned according to the method they target.
+			var verifiedSetupsPerMethod = new Dictionary<MethodInfo, List<Expression>>();
+
+			foreach (var setup in this.InterceptionContext.OrderedCalls)
 			{
-				var failures = calls.Values.Where(match);
-				if (failures.Any())
+				if (setup.IsConditional)
 				{
-					throw new MockVerificationException(failures.ToArray());
+					continue;
```

Comment:
Conditional setups were already excluded from `Mock.Verify[All]` prior to this PR, but that wasn't as easy to see as it is now (because previously, conditional setups only got added to `OrderedCalls`, but not to `calls`, and mass verification happened against the latter).